File: Leetcode 0657 Robot Return to Origin.py
# 2026/07/23
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        axis = [0,0]
        for i in moves:
            if i == "R":
                axis[0] += 1
            elif i == "L":
                axis[0] -= 1
            elif i == "U":
                axis[1] += 1
            elif i == "D":
                axis[1] -= 1

        return axis == [0,0]

File: Leetcode 0657 Robot Return to Origin.go
// 2026/07/23
func judgeCircle(moves string) bool {
    x := 0
    y := 0
    for i:=0; i<len(moves);i++ {
        switch s := string(moves[i]); s {
            case "U":
                x += 1
            case "D":
                x -= 1
            case "L":
                y -= 1
            case "R":
                y += 1
        }
    }
    if x ==0 && y == 0 {
        return true
    } else {
        return false
    }
}
